fhir-smart/src/test/java/com/ibm/fhir/smart/test/AuthzPolicyEnforcementTest.java

```diff
@@ -293,17 +293,17 @@ public void testBeforeSearch() throws FHIRPersistenceInterceptorException {
 
         // Invalid compartment search: not a Patient compartment
         try {
-            queryParameterValue.setValueString("Encounter/" + PATIENT_ID);
+            queryParameterValue.setValueString("Device/" + PATIENT_ID);
             properties.put(FHIRPersistenceEvent.PROPNAME_RESOURCE_TYPE, "Observation");
             properties.put(FHIRPersistenceEvent.PROPNAME_SEARCH_CONTEXT_IMPL, searchContext);
             FHIRPersistenceEvent event = new FHIRPersistenceEvent(observation, properties);
             interceptor.beforeSearch(event);
             fail("Patient interaction was allowed but should not be");
         } catch (FHIRPersistenceInterceptorException e) {
             // success
-            assertEquals(1, e.getIssues().size());
-            assertEquals(IssueType.FORBIDDEN, e.getIssues().get(0).getCode());
-            assertEquals("Compartment search with compartment 'Encounter' is not permitted.", e.getIssues().get(0).getDetails().getText().getValue());
+            assertEquals(e.getIssues().size(), 1);
+            assertEquals(e.getIssues().get(0).getCode(), IssueType.FORBIDDEN);
+            assertEquals(e.getIssues().get(0).getDetails().getText().getValue(), "Compartment search for compartment type 'Device' is not permitted.");
         }
 
         // Invalid compartment search: wrong Patient compartment
```

Comment:
Would be nice to have test for valid/invalid Encounter compartment search as well as valid List search.

Reply:
yes, I had this same thought but its a bit tricky because it requires a mock persistence layer that will respond to the read with an Encounter that is either in the Patient compartment (positive test) or not (negative test).  i'll see if i can whip something up quick...

Reply:
ok, i added the suggested test (and a couple more)